common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java

```diff
@@ -77,6 +75,24 @@ public MLAgent(String name,
         this.createdTime = createdTime;
         this.lastUpdateTime = lastUpdateTime;
         this.appType = appType;
+        validate();
+    }
+
+    private void validate() {
+        if (name == null) {
+            throw new IllegalArgumentException("agent name is null");
+        }
+        Set<String> toolNames = new HashSet<>();
+        if (tools != null) {
+            for (MLToolSpec toolSpec : tools) {
+                String toolName = Optional.ofNullable(toolSpec.getName()).orElse(toolSpec.getType());
+                if (toolNames.contains(toolName)) {
+                    throw new IllegalArgumentException("Duplicate tool defined: " + toolName);
```

Comment:
Is it expected to throw this exception or just remove the redundancy in the tools and return normally?

Reply:
Yes, it's expected, we can't remove redundancy for the user automatically as we don't know which one is the correct.